arch/arm/src/rp2040/rp2040_i2c.c

```diff
@@ -0,0 +1,832 @@
+ ****************************************************************************/
+
+static int rp2040_i2c_receive(struct rp2040_i2cdev_s *priv, int last)
+{
+  struct i2c_msg_s *msg = priv->msgs;
+  int i;
+  int en;
+  ssize_t msg_length;
+  irqstate_t flags;
+
+  priv->reg_buff_offset = 0;
+
+  DEBUGASSERT(msg != NULL);
+
+  for (msg_length = msg->length; msg_length > 0; msg_length -= priv->rw_size)
+    {
+      if (msg_length <= I2C_FIFO_MAX_SIZE)
+        {
+          priv->rw_size = msg_length;
+          en = 1;
+        }
+      else
+        {
+          priv->rw_size = I2C_FIFO_MAX_SIZE;
+          en = 0;
+        }
+
+      /* update threshold value of the receive buffer */
+
+      i2c_reg_write(priv, RP2040_I2C_IC_RX_TL_OFFSET, priv->rw_size - 1);
+
+      for (i = 0; i < priv->rw_size - 1; i++)
+        {
+          i2c_reg_write(priv, RP2040_I2C_IC_DATA_CMD_OFFSET,
+                              RP2040_I2C_IC_DATA_CMD_CMD);
+        }
+
+      flags = enter_critical_section();
+      wd_start(&priv->timeout, I2C_TIMEOUT,
+               rp2040_i2c_timeout, (wdparm_t)priv);
+
+      /* Set stop flag for indicate the last data */
+
+      i2c_reg_write(priv, RP2040_I2C_IC_DATA_CMD_OFFSET,
+                          RP2040_I2C_IC_DATA_CMD_CMD |
+                          (en ? RP2040_I2C_IC_DATA_CMD_STOP : 0));
+
+      i2c_reg_rmw(priv, RP2040_I2C_IC_INTR_MASK_OFFSET,
+                  RP2040_I2C_IC_INTR_STAT_R_RX_FULL,
+                  RP2040_I2C_IC_INTR_STAT_R_RX_FULL);
+      leave_critical_section(flags);
+      i2c_takesem(&priv->wait);
+
+      if (priv->error != OK)
+        {
+          break;
+        }
+    }
+
+  return 0;
+}
+
+/****************************************************************************
+ * Name: rp2040_i2c_send
+ *
+ * Description:
+ *   Send data to I2C bus.
+ *
+ ****************************************************************************/
+
+static int rp2040_i2c_send(struct rp2040_i2cdev_s *priv, int last)
+{
+  struct i2c_msg_s *msg = priv->msgs;
+  ssize_t i;
+  irqstate_t flags;
+
+  DEBUGASSERT(msg != NULL);
+
+  for (i = 0; i < msg->length - 1; i++)
+    {
+      while (!(i2c_reg_read(priv, RP2040_I2C_IC_STATUS_OFFSET)
+               & RP2040_I2C_IC_STATUS_TFNF))
+        ;
+
+      i2c_reg_write(priv, RP2040_I2C_IC_DATA_CMD_OFFSET,
+                    (uint32_t)msg->buffer[i]);
+    }
+
+  while (!(i2c_reg_read(priv, RP2040_I2C_IC_STATUS_OFFSET)
+           & RP2040_I2C_IC_STATUS_TFNF))
+    ;
+
+  flags = enter_critical_section();
+  wd_start(&priv->timeout, I2C_TIMEOUT,
+           rp2040_i2c_timeout, (wdparm_t)priv);
+  i2c_reg_write(priv, RP2040_I2C_IC_DATA_CMD_OFFSET,
+                (uint32_t)msg->buffer[i] |
+                (last ? RP2040_I2C_IC_DATA_CMD_STOP : 0));
+
+  /* Enable TX_EMPTY interrupt for determine transfer done. */
+
+  i2c_reg_rmw(priv, RP2040_I2C_IC_INTR_MASK_OFFSET,
+              RP2040_I2C_IC_INTR_STAT_R_TX_EMPTY, RP2040_I2C_IC_INTR_STAT_R_TX_EMPTY);
```

Comment:
Please break this line to put the last parameter in the new line, this way it will pass in the CI test

Reply:
Oops. I overlooked the style error before commit. Thanks.